agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -3628,4 +3594,88 @@ bool backhaul_manager::add_ap_he_capabilities(const sRadioInfo &radio_info)
     return true;
 }
 
+bool backhaul_manager::add_link_metrics(const sMacAddr &reporter_al_mac,
+                                        const sLinkInterface &link_interface,
+                                        const sLinkNeighbor &link_neighbor,
+                                        const sLinkMetrics &link_metrics,
+                                        ieee1905_1::eLinkMetricsType link_metrics_type)
+{
+    /**
+     * Add Transmitter Link Metric TLV if specifically requested or both requested
+     */
+    if ((ieee1905_1::eLinkMetricsType::TX_LINK_METRICS_ONLY == link_metrics_type) ||
+        (ieee1905_1::eLinkMetricsType::BOTH_TX_AND_RX_LINK_METRICS == link_metrics_type)) {
+        auto tlvTransmitterLinkMetric = cmdu_tx.addClass<ieee1905_1::tlvTransmitterLinkMetric>();
+        if (!tlvTransmitterLinkMetric) {
+            LOG(ERROR) << "addClass ieee1905_1::tlvTransmitterLinkMetric failed";
+            return false;
+        }
+
+        tlvTransmitterLinkMetric->reporter_al_mac() = reporter_al_mac;
+        tlvTransmitterLinkMetric->neighbor_al_mac() = link_neighbor.al_mac;
+
+        if (!tlvTransmitterLinkMetric->alloc_interface_pair_info()) {
+            LOG(ERROR) << "alloc_interface_pair_info failed";
+            return false;
+        }
+        auto interface_pair_info = tlvTransmitterLinkMetric->interface_pair_info(0);
+        if (!std::get<0>(interface_pair_info)) {
+            LOG(ERROR) << "Failed accessing interface_pair_info";
+            return false;
+        }
+        auto interfacePairInfo                      = std::get<1>(interface_pair_info);
+        interfacePairInfo.rc_interface_mac          = link_interface.iface_mac;
+        interfacePairInfo.neighbor_interface_mac    = link_neighbor.iface_mac;
+        interfacePairInfo.link_metric_info.intfType = link_interface.media_type;
+        // TODO
+        //Indicates whether or not the 1905.1 link includes one or more IEEE 802.1 bridges
+        //eIEEE802_1BridgeFlag IEEE802_1BridgeFlag;
```

Comment:
We do have to set that to something!

In fact, since we currently consider the bridge as the only interface on the local device for the wired backhaul, we should set the bridge flag to true for the wired link.

Reply:
I thought we had to implement IEEE 802.1 bridge discovery to be able to set this field appropriately and then set it to true if "the 1905.1 neighbor device is connected on this
particular interface through one or more IEEE 802.1 bridges". Are you sure this flag must be set to true for the wired link just because the wired backahul link is the bridge interface?

Reply:
I don't think it matters much either way, because I don't think anyone will use this. Probably leaving the TODO is the best option anyway. 

My idea was that this flag could be used to "know" that the topology is inaccurate (because there's a bridge in-between that doesn't participate in 1905.1, we don't have an accurate picture of the exact links). Since that is exactly the situation in our case (e.g. the controller may not be a direct neighbour, but still we treat it as one), I feel it is "safer" to set the bridge flag.